internal/consensus/state.go

```diff
@@ -1415,7 +1417,7 @@ func (cs *State) createProposalBlock(ctx context.Context) (*types.Block, error)
 
 	proposerAddr := cs.privValidatorPubKey.Address()
 
-	return cs.blockExec.CreateProposalBlock(ctx, cs.Height, cs.state, commit, proposerAddr, cs.LastCommit.GetVotes())
+	return cs.blockExec.CreateProposalBlock(ctx, cs.Height, cs.state, commit, proposerAddr, votes)
```

Comment:
So, if we are at `InitialHeight`, `votes` will be empty. Is this case taken into account inside `CreateProposalBlock` ? (In particular, when accessing the array inside the "for" loop)

Reply:
Technically, at `InitialHeight`, there's no way we could have vote extensions from a previous round, right?

Also, the `types.NewCommit` method sets `commitSigs` to `nil` (the last parameter in the constructor). When `CreateProposalBlock` is called, the `localLastCommit` struct ends up being empty (i.e. the `CommitInfo.Votes` slice will be empty). This means that the `for` loop won't loop over anything in `extendedCommitInfo`.